base/src/com/google/idea/blaze/base/command/mod/BlazeModRunnerImpl.java

```diff
@@ -38,17 +38,37 @@
 
 public class BlazeModRunnerImpl extends BlazeModRunner {
 
+  private static final String DUMP_REPO_MAPPING = "dump_repo_mapping";
+  private static final String ROOT_WORKSPACE = "";
+
+  /**
+   * {@code bazel mod dump_repo_mapping} takes a canonical repository name and will dump a map from
+   * repoName -> canonicalName of all the external repositories available to that repository The
+   * name {@code ""} is special and considered to be <em>the main workspace</em> so in order to dump
+   * the main repository map we would invoke it like {@code bazel mod dump_repo_mapping ""}.
+   *
+   * <p>Additionally the flag {@code --enable_workspace} needs to be off for this to work. The flag
```

Comment:
@mtoader do you remember what was the reason for this? Why enable workspace would break the call?

Reply:
It returns broken data. 

Basically you get all the dependencies visible from WORKSPACE and modules. Currently I pass it with `--noenable_workspace` to disable workspace and just look at the module provided dependencies. This is because the flag default value in 7+ is on. 

E.g.: a run with `--noenable_workspace` on my test repo (the correct run) 
```
➜ bazel mod --noenable_workspace dump_repo_mapping "" | yq -P | wc -l
      10
```
v.s. a run with `--enable_workspace` in the same spot:
```
➜ bazel mod --enable_workspace dump_repo_mapping "" | yq -P | wc -l
 checking cached actions
      93
```
all the extras look like:
```
remote_jdk8_linux: remote_jdk8_linux
bazel_skylib: bazel_skylib
remotejdk11_linux_toolchain_config_repo: remotejdk11_linux_toolchain_config_repo
remotejdk11_macos: remotejdk11_macos
remotejdk11_win_arm64: remotejdk11_win_arm64
remotejdk21_macos: remotejdk21_macos
rules_rust: rules_rust+
remotejdk17_macos_aarch64_toolchain_config_repo: remotejdk17_macos_aarch64_toolchain_config_repo
remote_jdk8_linux_aarch64: remote_jdk8_linux_aarch64
local_config_cc: local_config_cc
remotejdk11_macos_aarch64_toolchain_config_repo: remotejdk11_macos_aarch64_toolchain_config_repo
remotejdk21_linux: remotejdk21_linux
android_gmaven_r8: android_gmaven_r8
remote_java_tools_linux: remote_java_tools_linux
remotejdk21_win: remotejdk21_win
```
aka bazel toolchain variants.

Reply:
oh, ok, thank you!

Reply:
I asked, because I had to introduce `bazel mod deps` call, and it crashes with --noenable_workspace, due to a bug in bazel 7.3. So I was wondering what was the purpose in dump_repo_mapping